### PR TITLE
Furnace and Shrike lathe upgrades

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
This one's for the miner gang. 
To start with, furnaces are now able to be built and deconstructed. Their boards have been added to the basic mining technology techweb node. They can also be upgraded for more mining points and faster performance. In addition, they can be set to smelt only a certain amount of material instead of all of a type to better allow people to use their infinite storage of unsmelted ores.
Finally, the lathes on the Shrike class vessels have been given a capacity upgrade and can hold significantly more resources than standard unlinked lathes. The storage will still scale with part upgrades.
I didn't feel like making the console detect walls intelligently though.

## Why It's Good For The Game
Makes alternate mining equipment more fully-featured and addresses some issues with silo-less mineral storage.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://streamable.com/g2cgz3

</details>

## Changelog
:cl:
balance: Increased Shrike base protolathe/techfab storage capacity
add: Added circuitboards for the furnace and furnace control console to mining tech research
add: Added construction, deconstruction, and multitool linking of furnace and furnace console
add: Added stock parts upgrades for the furnace
fix: Fixed mining points not appearing on the smelter console after the TGUI upgrade
/:cl:

### CODE REVIEW

File: _maps/map_files/Shrike/job_changes.dm [hunk 1]
```diff
@@ -98,6 +98,11 @@ MAP_REMOVE_JOB(air_traffic_controller)
 		return
 	hidden = FALSE
 
+// Lathe storage
+/obj/machinery/rnd/production/New()
+	. = ..()
+	base_storage *= 5
```
Comment (Karmic-Skink) [inline _maps/map_files/Shrike/job_changes.dm:104]: May as well have an actual ore sile and ORM with this modifier combined with the infinite storage in the smelter.

Reduce the multiplier to 2x max.

**Review by Karmic-Skink — DISMISSED** [review]

File: _maps/map_files/Shrike/job_changes.dm (hunk 1 above)
Comment (Karmic-Skink) [reply _maps/map_files/Shrike/job_changes.dm:104]: Does the smelter in this PR not have buttons for dispensing exact materials on demand?
Why would you want to move it to an additional device?

**Review by Bokkiewokkie — DISMISSED** [review]
Code looks good, but here comes the part where I ask you to port it to Beestation
👉👈

File: _maps/map_files/Shrike/job_changes.dm (hunk 1 above)
Comment (Bokkiewokkie) [reply _maps/map_files/Shrike/job_changes.dm:104]: 5 does seem like a lot, we could always just make it *2 and scale it up if that's not enough

**Review by Bokkiewokkie — DISMISSED** [review]
2

**Pull request merged**